### DISCUSSION TITLE
.obj import .glb (glb2) export

### DISCUSSION THREAD

**Discussion by jonathanperezg** [assimp/assimp#4271] [Q&A]
Hello everyone, my goal is to convert an obj file (along with its corresponding mtl file) to glb2. 

The approach I'm currently taking is to import the obj file using the Assimp::Importer::Importer, and exporting using the Assimp::Exporter::Exporter. 

I am currently getting lots of errors from the glTF validator. Screen shot is below.

Any help would be appreciated. Below is all the code I'm using to do this:

``` 
#include <iostream>
#include <string>
#include <assimp/Importer.hpp> 
#include <assimp/Exporter.hpp> 
#include <assimp/scene.h>
#include <assimp/postprocess.h>


using namespace std;

int main(){

    const string objFile = ("./duck.obj");
    const string outputFile = ("./duck.glb");
    const string FormatId = ("glb2");
    Assimp::Importer importer;
    Assimp::Exporter exporter;
    const aiScene* scene = importer.ReadFile(objFile, 0);
    
    if (scene){
        cout << scene->HasMeshes() << endl;
        
        cout << scene->HasMaterials() << endl;

        aiReturn ret = exporter.Export(scene, FormatId, outputFile);

        if (ret == AI_SUCCESS) {
            cout << "The file was exported successfully!" << endl;
        }
        else{
            cout << "ERROR: The file was not exported successfully!" << endl;
        }

    }
    else{
        cout << importer.GetErrorString() << endl;
    }
    
    return 0;
} 
```
![image](https://user-images.githubusercontent.com/37051940/147008974-e58c16bd-0c69-40e7-b15c-595f2a521f4b.png)
![image](https://user-images.githubusercontent.com/37051940/147009089-939c21df-85fe-4b06-8030-6fe0d180a53c.png)

**Comment by kimkulling** [comment]
Looks pretty bad. I guess the export for glTF2 models is broken.

**Comment by gamersi** [comment]
Is there any update, I am having the same issue

**Comment by kimkulling** [comment]
No, not now. We made some progress but not all issues are fixed and validated.
